packages/ui/SingleResponseCard/actions.ts

```diff
@@ -23,12 +24,20 @@ export const deleteResponseAction = async (responseId: string) => {
 export const updateResponseNoteAction = async (responseNoteId: string, text: string) => {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
+
+  const isAuthorized = await canUserAccessResponseNote(session.user!.id, responseNoteId);
+  if (!isAuthorized) throw new AuthorizationError("Not authorized");
+
   await updateResponseNote(responseNoteId, text);
 };
 
 export const resolveResponseNoteAction = async (responseNoteId: string) => {
```

Comment:
@ShubhamPalriwala everyone from the team should be able to resolve a response note :-)

Reply:
have made the change, please take a look now. we now have 2 separate auth methods to follow the least privilege principle ie 
- can user modify response note (for response creators)
- can use resolve response note (for access to the response note and its core team)

Reply:
I went ahead with 2 different methods is because:
- a responseNote should be edited/deleted only by the user who created it
- a responseNote should be resolved by the user who has access to the team inside which the responseNote exists.


And as you said, would you prefer something that just checks if the user has access to the team that has the responseNote (and then we leave the micro authzn checks of editing privileges)?

Reply:
I thought that we can just do it like this:
```ts
export const resolveResponseNoteAction = async (responseId: string, responseNoteId: string) => {
  const session = await getServerSession(authOptions);
  if (!session) throw new AuthorizationError("Not authorized");

  const isAuthorized = await canUserAccessResponse(session.user!.id, responseId);
  if (!isAuthorized) throw new AuthorizationError("Not authorized");

  await resolveResponseNote(responseNoteId);
};
```
(without the extra `canUserResolveResponseNote()` function)